google/cloud/storage/examples/storage_bucket_autoclass_samples.cc

```diff
@@ -0,0 +1,139 @@
+                         gcs::BucketAutoclass{enabled}));
+    if (!metadata) throw google::cloud::Status(std::move(metadata).status());
+
+    std::cout << "The autoclass configuration for bucket " << bucket_name
+              << " was successfully updated.";
+    if (!metadata->has_autoclass()) {
+      std::cout << " The bucket no longer has an autoclass configuration.\n";
+      return;
+    }
+    std::cout << " The new configuration is " << metadata->autoclass() << "\n";
+  }
+  //! [set-autoclass] [END storage_set_autoclass]
+  (std::move(client), argv.at(0), enabled);
+}
+
+void RunAll(std::vector<std::string> const& argv) {
+  namespace examples = ::google::cloud::storage::examples;
+  namespace gcs = ::google::cloud::storage;
+
+  if (!argv.empty()) throw examples::Usage{"auto"};
+  examples::CheckEnvironmentVariablesAreSet({
+      "GOOGLE_CLOUD_PROJECT",
+  });
+  auto const project_id =
+      google::cloud::internal::GetEnv("GOOGLE_CLOUD_PROJECT").value();
+  auto generator = google::cloud::internal::DefaultPRNG(std::random_device{}());
+  auto const bucket_name_enabled = examples::MakeRandomBucketName(generator);
+  auto const bucket_name_disabled = examples::MakeRandomBucketName(generator);
+  auto const object_name =
+      examples::MakeRandomObjectName(generator, "object-") + ".txt";
+  auto client = gcs::Client();
+
+  std::cout << "\nCreating buckets to run the example:"
+            << "\nEnabled Autoclass: " << bucket_name_enabled
+            << "\nDisabled Autoclass: " << bucket_name_disabled << std::endl;
+  // In GCS a single project cannot create or delete buckets more often than
+  // once every two seconds. We will pause until that time before deleting the
+  // bucket.
+  auto constexpr kBucketPeriod = std::chrono::seconds(2);
+  auto pause = std::chrono::steady_clock::now() + kBucketPeriod;
+  (void)client
+      .CreateBucketForProject(
+          bucket_name_enabled, project_id,
+          gcs::BucketMetadata{}.set_autoclass(gcs::BucketAutoclass{true}))
+      .value();
+  if (!examples::UsingEmulator()) std::this_thread::sleep_until(pause);
+  pause = std::chrono::steady_clock::now() + kBucketPeriod;
+  (void)client
+      .CreateBucketForProject(
+          bucket_name_disabled, project_id,
+          gcs::BucketMetadata{}.set_autoclass(gcs::BucketAutoclass{false}))
+      .value();
+
+  std::cout << "\nRunning GetAutoclass() example [enabled]" << std::endl;
+  GetAutoclass(client, {bucket_name_enabled});
+
+  std::cout << "\nRunning GetAutoclass() example [disabled]" << std::endl;
+  GetAutoclass(client, {bucket_name_disabled});
+
+  std::cout << "\nRunning SetAutoclass() example" << std::endl;
+  SetAutoclass(client, {bucket_name_enabled, "false"});
+
+  if (!examples::UsingEmulator()) std::this_thread::sleep_until(pause);
+  (void)examples::RemoveBucketAndContents(client, bucket_name_enabled);
+  (void)examples::RemoveBucketAndContents(client, bucket_name_disabled);
```

Comment:
do you want a pause in between these deletes?

Reply:
Done.